crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -244,18 +246,19 @@ impl CoinReadApiServer for CoinReadApi {
                 debug!(?owner, "Failed to get all balance with error: {:?}", e);
                 Error::from(e)
             })?;
-        Ok(all_balance
-            .iter()
-            .map(|(coin_type, balance)| {
-                Balance {
-                    coin_type: coin_type.to_string(),
-                    coin_object_count: balance.num_coins,
-                    total_balance: balance.balance,
-                    // note: LockedCoin is deprecated
-                    locked_balance: Default::default(),
-                }
+        let mut result = vec![];
+        for (coin_type, balance) in all_balance.iter() {
+            result.push(Balance {
+                coin_type: coin_type.to_string(),
+                coin_object_count: usize::try_from(balance.num_coins)
+                    .map_err(|e| Error::SuiError(SuiError::ExecutionError(e.to_string())))?,
+                total_balance: u128::try_from(balance.balance)
+                    .map_err(|e| Error::SuiError(SuiError::ExecutionError(e.to_string())))?,
+                // note: LockedCoin is deprecated
```

Comment:
ditto.

The merge-op thing has a lot of integer conversions and I think unwrap may be better because the max balance of our Coin type is u64 so it should't be possible for this to fail. And if it fails, it'd better fail fast and scream

Reply:
sure, done